source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -1066,56 +1038,29 @@ ShaderNode* ShaderGraph::createNode(const Node& node, GenContext& context)
             _document->getActiveColorSpace() : context.getOptions().targetColorSpaceOverride;
     }
 
-    if (colorManagementSystem && !targetColorSpace.empty())
-    {
-        for (InputPtr input : node.getInputs())
-        {
-            ShaderInput* shaderInput = newNode->getInput(input->getName());
-            populateColorTransformMap(colorManagementSystem, shaderInput, input, targetColorSpace, true);
-        }
-        for (auto parameter : node.getParameters())
-        {
-            if (parameter->getType() == FILENAME_TYPE_STRING)
-            {
-                ShaderOutput* shaderOutput = newNode->getOutput();
-                if (shaderOutput)
-                {
-                    populateColorTransformMap(colorManagementSystem, shaderOutput, parameter, targetColorSpace, false);
-                }
-            }
-            else
-            {
-                ShaderInput* shaderInput = newNode->getInput(parameter->getName());
-                populateColorTransformMap(colorManagementSystem, shaderInput, parameter, targetColorSpace, true);
-            }
-        }
-    }
-
-    // Unit Conversion: Only applicable if Unit system and a "targetDistanceUnit" is defined for now.
     UnitSystemPtr unitSystem = context.getShaderGenerator().getUnitSystem();
     const string& targetDistanceUnit = context.getOptions().targetDistanceUnit;
-    if (unitSystem && !targetDistanceUnit.empty())
+
+    for (InputPtr input : node.getInputs())
     {
-        for (InputPtr input : node.getInputs())
-        {
-            ShaderInput* inputPort = newNode->getInput(input->getName());
-            populateUnitTransformMap(unitSystem, inputPort, input, targetDistanceUnit, true);
-        }
-        for (auto parameter : node.getParameters())
+        // It is sufficient that the input type is a filename regardless of whether it is marked as a uniform 
+        if (input->getType() == FILENAME_TYPE_STRING)
         {
-            ShaderInput* inputPort = newNode->getInput(parameter->getName());
-            if (parameter->getType() == FILENAME_TYPE_STRING)
+            ShaderOutput* shaderOutput = newNode->getOutput();
+            if (shaderOutput)
             {
-                // Check output port for any parameters which are file names
-                ShaderOutput* shaderOutput = newNode->getOutput();
-                if (shaderOutput)
-                {
-                    populateUnitTransformMap(unitSystem, shaderOutput, parameter, targetDistanceUnit, false);
-                }
+                populateColorTransformMap(colorManagementSystem, shaderOutput, input, targetColorSpace, false);
+                populateUnitTransformMap(unitSystem, shaderOutput, input, targetDistanceUnit, false);
             }
-            populateUnitTransformMap(unitSystem, inputPort, parameter, targetDistanceUnit, true);
```

Comment:
Fixed unit issue: Moved this to proper logic that we map input vs output, not map both input **and** output. Also collapsed double loop to single loop. The called methods do checks to determine applicability.